app/controllers/samples_controller.rb

```diff
@@ -138,14 +139,19 @@ def index_v2
                  current_power.samples
                end
 
-    respond_to do |format|
-      format.json do
-        render json: @samples.offset(offset).limit(limit).as_json(
-          only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at],
-          methods: []
-        )
-      end
+    @samples = @samples.offset(offset).limit(limit)
+
+    # Efficiently fetch auxiliary information from associated models
+    samples_resp = []
+    @samples.includes(:project, :host_genome, :input_files).each do |s|
+      # TODO: sample_tissue column is deprecated, use Metadatum model instead
+      entry = s.as_json(only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at])
+      entry["host_genome_name"] = s.host_genome_name # this method uses s.host_genome under the hood
+      entry["private_until"] = s.private_until # this method uses s.project under the hood
+      entry["input_files"] = s.input_files.as_json # TODO: can probably be removed, but I ported it from the legacy code just in case
```

Comment:
As you note, we do not need `input_files` here, specially since index_v2 is currently only use in discovery view.
As we migrate other calls to index_v2, we can do another request for input files from other specific endpoints.

Reply:
Thanks for confirming, I will remove